Mount the parent dir in daemonset configuration for openshift-sdn

abhat opened the pull request:
In the daemonset configuration for openshift-sdn the hostpath mount
specified is a file name instead of the parent dir. Kubernetes will
create the file but it will create it as a directory during pod setup.

This PR changes the hostpath mount to the parent dir and if a directory
by the name of origin-node exists, deletes it as a precaution.

abhat commented:
/assign @squeed

abhat commented:
/cc @squeed

squeed commented on roles/openshift_sdn/files/sdn.yaml:
```diff
@@ -77,6 +77,12 @@ spec:
           done
 
+          #BUG 1728081: cdc accidentally mounted /etc/sysconfig/origin-node as DirectoryOrCreate; clean it up so we can ultimately mount /etc/sysconfig/origin-node as FileOrCreate
+          # Once this is released, then we can mount it properly
```
We're never going to undo this, so this comment is misleading.

squeed commented on roles/openshift_sdn/files/sdn.yaml:
```diff
@@ -77,6 +77,12 @@ spec:
             fi
           done
 
+          #BUG 1728081: cdc accidentally mounted /etc/sysconfig/origin-node as DirectoryOrCreate; clean it up so we can ultimately mount /etc/sysconfig/origin-node as FileOrCreate
```
I didn't do this :-) I only broke 4.1.

squeed commented:
two small comment updates, otherwise fine.

abhat commented:
@squeed addressed the code review comments. I haven't been able to dev-test the PR yet because of some issue with 3.11 installer timing out on endpoints. Will update here when I get past that.

squeed commented on roles/openshift_sdn/files/sdn.yaml:
```diff
@@ -77,16 +77,24 @@ spec:
+          #BUG 1728081: If /etc/sysconfig/origin-node got mounted as DirectoryOrCreate; clean it up so we can ultimately mount /etc/sysconfig/origin-node as FileOrCreate
+          if [[ -d /etc/sysconfig/origin-node ]]; then
+            echo "Deleting the origin-node dir"
+            rmdir /etc/sysconfig/origin-node
```
can you do `rmdir ... || true` - if this somehow fails, we shouldn't take networking down on the node.

squeed commented on roles/openshift_sdn/files/sdn.yaml:
```diff
@@ -77,16 +77,24 @@ spec:
           # Load DEBUG_LOGLEVEL
           if [[ -f /etc/sysconfig/origin-node ]]; then
             set -o allexport
+            echo "Reading the options from origin-node directory"
```
"Reading environment variables from /etc/sysconfig/origin-node" (and similar below)

squeed commented:
/lgtm

abhat commented:
/assign @patrickdillon

patrickdillon commented:
/approve
